intercom_tag_users_block.py

```diff
@@ -0,0 +1,46 @@
+from nio.block.base import Block
+from nio.signal.base import Signal
+from nio.properties import IntProperty, StringProperty, ObjectProperty, \
+    PropertyHolder
+import requests
+
+
+class IntercomTagUsers(Block):
+
+    access_token = StringProperty(
+        title="Access Token", default="[[INTERCOM_ACCESS_TOKEN]]")
+    tag_name = StringProperty(title="Name of Tag", default="CrankyCustomer")
+    email = StringProperty(title="User Email", default="{{ $user }}")
+    user_id = StringProperty(
+        title="User ID", default="{{ $user_id }}", allow_none=True)
+
+    def configure(self, context):
+        super().configure(context)
+
+    def process_signals(self, signals):
+        for signal in signals:
+            response = self._request('post', body={
+                "name": self.tag_name(signal),
+                "users": [
+                    {"id": self.user_id(signal)},
+                    {"email": self.email(signal)}
+                ] if self.user_id(signal) else [{"email": self.email(signal)}],
```

Comment:
Lets talk about this part. I'm not seeing why `id` is even there. If `email` is required, why would you also put `id`? To me, it would make sense to make `email` be the only option (which I'm totally ok with) or make it something where you just need one or the other.